### PR TITLE
oem: Add Vultr as a provider

### PR DESCRIPTION
**Pull request by vbatts**
* added vultr to oem reg

cc @eb3095
ref https://github.com/eb3095/ignition/pull/2

Co-authored-by: Mike Biondi <biondizzle@gmail.com>
Co-authored-by: eb3095 <45504889+eb3095@users.noreply.github.com>
Signed-off-by: Vincent Batts <vbatts@kinvolk.io>

# oem: Add Vultr as a provider

# How to use

`./bin/amd64/ignition -oem vultr ...`

# Testing done

Have not

### CODE REVIEW

File: internal/providers/vultr/vultr.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The vultr provider fetches a remote configuration from the vultr
+// user-data metadata service URL.
+// https://web.archive.org/web/20190513194756/https://www.vultr.com/metadata/#user
```
Comment (pothos) [inline internal/providers/vultr/vultr.go:17]: This link says that user data is under `/user-data/user-data`, not `latest/user-data` – are both correct?

Comment (pothos) [reply internal/providers/vultr/vultr.go:17]: Yeah, that's also something already upstream: https://github.com/coreos/ignition/blob/master/internal/providers/vultr/vultr.go
That uses `/user-data/user-data`

**Comment by eb3095** [comment]
Yes both are infact correct. The first end is a regex.

**Comment by vbatts** [comment]
:+1:

**Review by pothos — APPROVED** [review]

**Pull request merged**